lib/coap/connection.ex

```diff
@@ -267,7 +319,27 @@ defmodule CoAP.Connection do
   # DELIVER ====================================================================
   # reply from app to peer
```

Comment:
Just to verify my understanding, `app == server` and `peer == client`?

Reply:
Yeah. There's a lot of terms that end up being nearly synonymous:

`app` vs `peer`
`server` vs `client`
`receiver` vs `requester`
`request` vs `response`

This is all somewhat complicated by the stateful connection, because UDP. It isn't a response on the same TCP as with HTTP, it's almost as a "request" in the opposite direction over a known set of IP+PORT combos.

I'd like to add a glossary of how we determine which is which. For example. Request/response can (appears to) consistently be determined by the `code`, it's either a verb (0) or status (>0).

Reply:
To answer your question. Not exactly 😛. It depends on if you are the client or the server.

`app` is the code that handles the logic outside of the connection state. On the server, that's like phoenix. On the client it's just a simple module + the library user's logic.

`peer` is whoever is on the other end of the UDP connection.

It's tough, as the `connection` actually sits between the `app` and the `peer`. The `app` is "local" and the `peer` is over UDP.